lecture_2/homework/app/models/warrior.rb

```diff
@@ -0,0 +1,18 @@
+class Warrior < ApplicationRecord
+    belongs_to :clan
+    belongs_to :structure
+    belongs_to :weapon
+
+    validates :name, presence: true, uniqueness: { scope: :clan_id }, if: Proc.new { |it| it.death_date.blank? }, on: :update
+    validates :name, presence: true, uniqueness: { scope: :clan_id }, unless: Proc.new { |it| it.death_date.blank? }, on: :create
```

Comment:
Ta walidacja nie do końca działa tak jak chcemy. Jeżeli będziemy tworzyć nowego wojownika, który nazywa się tak jak jeden z martwych to dostaniemy błąd z bazy danych (bo ta walidacja zostanie pominięta). Popatrz jakie opcje poza `scope` akceptuje `uniqueness` i spróbuj tutaj użyć jednej z nich

Reply:
Po poprawkach następujący kod powinien zwrócić błąd walidacji a nie błąd z bazy danych:
```ruby
h = Hussar.create!(name: 'Foo', clan: Clan.last, structure: Structure.last, weapon: Weapon.last)
h2 = h.dup
h2.save! # error
```

Reply:
Poprawiłem walidację, ale nie mam 100% pewności, czy o to chodziło. Będę wdzięczny, jak rzucisz okiem na poprawkę.

Reply:
Dokładnie o to chodziło, teraz wszystko działa jak należy. Warunek `where('death_date IS NULL')` można też zapisać bez wstawiania sqla: `where(death_date: nil)`, ale to już taka uwaga stylistyczna 😃

Reply:
Elegancko :-) Przy okazji, zamieniłem tę SQLkę w walidacji i w scope'ach, dzięki za sugestię!